tests/phpunit/tests/test-wpml-cornerstone-update-translation.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/**
+ * Class Test_WPML_Cornerstone_Update_Translation
+ *
+ * @group cornerstone
+ */
+class Test_WPML_Cornerstone_Update_Translation extends OTGS_TestCase {
+
+	/**
+	 * @test
+	 */
+	public function it_updates_translation() {
+		\WP_Mock::wpPassthruFunction( '__' );
+		$translated_post_id        = mt_rand();
+		$original_post_id          = mt_rand();
+		$original_post             = (object) array( 'ID' => $original_post_id );
+		$lang                      = 'en';
+		$translation               = 'translation-value';
+
+		$nodes = array(
+			'custom-node' => array(
+				'conditions'        => array( '_type' => 'custom-node' ),
+				'fields'     => array(
+					array(
+						'field'       => 'text',
+						'type'        => __( 'Custom node text', 'sitepress' ),
+						'editor_type' => 'LINE'
+					),
+				),
+			),
+		);
+
+		$meta_field_data = array(
+			array(
+				'_type' => 'custom-node',
+				'text' => 'value',
+			),
+		);
+
+		$meta_field_translated_data = array(
+			array(
+				'_type' => 'custom-node',
+				'text' => $translation,
+			),
+		);
+
+		$node_id = md5( serialize( $meta_field_data[0] ) );
+
+		$string_translations = array(
+			'text-custom-node-' . $node_id => array(
+				$lang => array(
+					'status' => 10,
+					'value'  => $translation
+				)
+			)
+		);
+
+		\WP_Mock::wpFunction( 'get_post_meta', array(
+			'times'  => 1,
+			'args'   => array( $original_post_id, '_cornerstone_data', true ),
+			'return' => $meta_field_data,
+		) );
+
+		\WP_Mock::wpFunction( 'update_post_meta', array(
+			'times' => 1,
+			'args'  => array( $translated_post_id, '_cornerstone_data', $meta_field_translated_data ),
+		) );
+
+		$this->add_copy_meta_fields_checks( $translated_post_id, $original_post_id );
+
+		$translatable_nodes = new WPML_Cornerstone_Translatable_Nodes();
+
+		$data_settings = $this->getMockBuilder( 'WPML_Cornerstone_Data_Settings' )
+		                      ->disableOriginalConstructor()
+		                      ->getMock();
+		$data_settings->method( 'get_meta_field' )
+		              ->willReturn( '_cornerstone_data' );
+		$data_settings->method( 'get_node_id_field' )
+		              ->willReturn( 'id' );
+		$data_settings->method( 'get_fields_to_copy' )
+		              ->willReturn( array( '_cornerstone_settings', '_cornerstone_version' ) );
+		$data_settings->method( 'convert_data_to_array' )
+		              ->with( $meta_field_data )
+		              ->willReturn( $meta_field_data );
+		$data_settings->method( 'prepare_data_for_saving' )
+		              ->with( $meta_field_translated_data )
+		              ->willReturn( $meta_field_translated_data );
+		$data_settings->method( 'get_fields_to_save' )
+		              ->willReturn( array( '_cornerstone_data' ) );
+
+		\WP_Mock::onFilter( 'wpml_cornerstone_modules_to_translate' )
+			->with( $this->get_translatable_nodes() )
+			->reply( $nodes );
```

Comment:
As `WPML_Cornerstone_Update_Translation` is very connected to `WPML_Cornerstone_Translatable_Nodes`, I decided to include a real object of `WPML_Cornerstone_Translatable_Nodes` so I can get real results when using the methods `get` and `update`. So, this test here is also (not directly) testing `WPML_Cornerstone_Translatable_Nodes`.

Also, this `\WP_Mock::onFilter( 'wpml_cornerstone_modules_to_translate' )` is working mostly like an assertion, I chose this approach because If any translatable node is added in `WPML_Cornerstone_Translatable_Nodes` then this line here will throw an error until developer includes the node in the list of translatable nodes in this test which will satisfy this filter assertion then replying with the custom `$node` added here.